tests/std/tests/P0645R10_text_formatting_custom_formatting/test.cpp

```diff
@@ -59,6 +69,20 @@ struct std::formatter<basic_custom_formattable_type, char> {
     }
 };
 
+template <>
+struct std::formatter<move_only_custom_formattable_type, char> {
+    basic_format_parse_context<char>::iterator parse(basic_format_parse_context<char>& parse_ctx) {
+        if (parse_ctx.begin() != parse_ctx.end()) {

```

Comment:
yeah, this format method is bogus, but "}" is never the spec string, it's that [pc.begin(), pc.end()] is from the start of the format spec to the end of _the entire format string_, so finding '}' is just the end of the spec

(why yes, this does allow you to "eat" replacement fields in formatters in theory, but that's the entire point, and the parser will blow up immediately with format_error (or a compile failure) if you try and do that, what it allows is stuff like `{:}aaaa}bbbbbbb` where `}aaaa` is the format specifier for the first (and only) argument. it should also allow nesting, for example
`
std::format(R"(
{:
int main() {
   std::string hello = "hello";
   fputs(stdout, std::format("{:*20}\n", hello);
   return 5;
}
}
)", some_jit_or_interpreter_context{});
`
could return something "reasonable"

Reply:
the begin()==end() condition is "no format specifiers", in this case begin==end and they are null iterators (you get a default initialized parse context)

Reply:
so, actually I think this is a _valid_ formatter, because it throws an error when the whole range is not parsed (since no format specifiers are allowed)